packages/angular_devkit/build_webpack/src/browser/index.ts

```diff
@@ -0,0 +1,258 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { BuildEvent, Builder, BuilderContext, Target } from '@angular-devkit/architect';
+import { getSystemPath } from '@angular-devkit/core';
+import { writeFileSync } from 'fs';
+import { resolve } from 'path';
+import { Observable } from 'rxjs/Observable';
+import * as ts from 'typescript'; // tslint:disable-line:no-implicit-dependencies
+import * as webpack from 'webpack';
+import {
+  getAotConfig,
+  getBrowserConfig,
+  getCommonConfig,
+  getDevConfig,
+  getNonAotConfig,
+  getProdConfig,
+  getStylesConfig,
+} from '../angular-cli-files/models/webpack-configs';
+import { getWebpackStatsConfig } from '../angular-cli-files/models/webpack-configs/utils';
+import { readTsconfig } from '../angular-cli-files/utilities/read-tsconfig';
+import { requireProjectModule } from '../angular-cli-files/utilities/require-project-module';
+import {
+  statsErrorsToString,
+  statsToString,
+  statsWarningsToString,
+} from '../angular-cli-files/utilities/stats';
+const webpackMerge = require('webpack-merge');
+
+
+// TODO: Use quicktype to build our TypeScript interfaces from the JSON Schema itself, in
+// the build system.
+export interface BrowserBuilderOptions {
```

Comment:
Should we attempt standardize a minimal set of _browser_ options that all browser builds should support (or indicate that they support)?  This should ideally help with transitioning between different builders.

Or maybe some type of feature flagging capability that allows for a standard way to configure certain options (or option sets)?  Prime possibilities could be assets, external scripts, or i18n.

Reply:
I tried to follow that train of thought for a while a month of so ago and didn't think we could hope to do it. Every setup uses different names for everything. And it wouldn't make much of a difference overall, because browser builds are just one of the many we need to do.

Reply:
Well underneath the names could be different, but I think a big selling point of a facade is that it hides differences between the underlying implementations (i.e., standardized nomenclature and concepts on the user facing side with adapters and transformations on the inside).